```
CMake Warning at test/CMakeLists.txt:97 (message):
  Minimum cmake version of 3.12 not met for Google benchmark!


-- Could NOT find GTest (missing: GTEST_LIBRARY GTEST_MAIN_LIBRARY)
CMake Error at test/CMakeLists.txt:127 (include):
  include could not find load file:

    FetchContent


-- Git checking out GoogleTest release-1.12.1
CMake Error at test/CMakeLists.txt:148 (FetchContent_Declare):
  Unknown CMake command "FetchContent_Declare".


-- Configuring incomplete, errors occurred!
```

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -94,16 +94,11 @@ endif()
 
 if(WITH_GTEST OR WITH_BENCHMARKS)
     if(CMAKE_VERSION VERSION_LESS 3.12)
-        message(WARNING "Minimum cmake version of 3.12 not met for GoogleTest or benchmarks!")
-
-        set(WITH_GTEST OFF)
-        set(WITH_GTEST OFF PARENT_SCOPE)
-
+        message(WARNING "Minimum cmake version of 3.12 not met for Google benchmark!")
         set(WITH_BENCHMARKS OFF)
         set(WITH_BENCHMARKS OFF PARENT_SCOPE)
-    else()
-        enable_language(CXX)
     endif()
+    enable_language(CXX)
 endif()
 
 if(WITH_BENCHMARKS)
@@ -127,7 +122,7 @@ if(WITH_GTEST)
         find_package(GTest)
     endif()
 
-    if(NOT TARGET GTest::GTest)
+    if(NOT TARGET GTest::GTest AND NOT CMAKE_VERSION VERSION_LESS 3.11)
         include(FetchContent)
 
         # Prevent overriding the parent project's compiler/linker settings for Windows

